LibreNMS/queuemanager.py

```diff
@@ -537,6 +537,56 @@ def __init__(self, config, lock_manager):
 
     def do_work(self, device_id, group):
         if self.lock(device_id, timeout=self.config.poller.frequency):
+            # We need to check if the device was polled recently to prevent a race condition
+            # We have 3 cases:
+            # 1. The last_polled_timetaken is greater than the poller frequency, we can poll the device
+            # The last_polled_timetaken is less than the poller frequency cap to maximum 15 seconds
+            #   2. If the last_polled is greater than the cap, we can poll the device
+            #   3. If the last_polled is less than the cap, we skip polling the device
+            dbPollerQueueManager = LibreNMS.DB(self.config)
```

Comment:
Isn't this making a new db connection every 15 seconds?

Reply:
Yes, it's a trade off to close the connections ASAP to avoid too many connections.
As in a 30workers/10pollers setup it would have 300 connections never releasing/idle.

Reply:
I dont understand why the locking would not work with Redis. Does this have something to do with the Sentinel like the local nodes are somehow out of sync? When reading the docs, I see that there is asynchronous replication mentioned which might cause the data on invidual Redis servers to be out of sync for a while...

Reply:
I will test this on my installation today.

Reply:
I still wonder if this is fixing the issue only by adding delay.

Reply:
> I still wonder if this is fixing the issue only by adding delay.

Probably. Just checked logs with this patch and indeed, there are thousands of skips "...polled recently" entries now.

Should probably use some real queuing system which is managed by the master node. The master would queue tasks (polls, discoveries, retries) and keep track of the task statuses. Isnt this exactly what the @Jellyfrog 's Laravel queue patch was trying to do?